### PR TITLE
CP-44441: Restore SNMP service config

### PR DESCRIPTION
**Pull request by DeliZhangX**
In versions prior to XS 8.4, the net-snmp rpm already existed, but we did not declare to support this feature. Now in XS 8.4 we have enhanced this feature and will officially release it. XS 8.4 is the first version to support the SNMP feature. The host-installer is modified here to automatically back up and restore the corresponding configuration files when RPU to future versions. Versions prior to 8.4 do not require backup and restoration.

### CODE REVIEW

**Comment by freddy77** [comment]
Please provide a description

File: upgrade.py [hunk 1]
```diff
@@ -448,6 +448,15 @@ def buildRestoreList(self):
         # NRPE service config
         self.restore_list += ['etc/nagios/nrpe.cfg', {'dir': 'etc/nrpe.d'}]
 
+        # SNMP service config
```
Comment (freddy77) [inline upgrade.py:451]: Looking at other similar code in the same function many comments refer to the Jira ticket (like `# CA-195388: Preserve /etc/mdadm.conf across upgrades` above). Could you add such reference?

Comment (DeliZhangX) [reply upgrade.py]: Thanks for your remind. I added PR description and jira ticket number in comment.

**Review by freddy77 — APPROVED** [review]

**Comment by liulinC** [comment]
@DeliZhangX  it this required for xs9? if yes, do not forget a following up PR to master.

**Comment by DeliZhangX** [comment]
May I ask who can help to merge?

**Comment by freddy77** [comment]
> May I ask who can help to merge?

Done. About the merge in master cherry-pick the commit on master and open a new PR.

**Comment by DeliZhangX** [comment]
> > May I ask who can help to merge?
> 
> Done. About the merge in master cherry-pick the commit on master and open a new PR.

Raised https://github.com/xenserver/host-installer/pull/97

**Pull request merged**